src/lib/modules/blockchain_connector/provider.js

```diff
@@ -16,9 +20,31 @@ class Provider {
 
   startWeb3Provider(callback) {
     const self = this;
+    const accountsConfig = self.accountsConfig[0];
```

Comment:
Can you put this variable inside the `else if (this.type === 'ledger') {` as it is the only place where it is used.

Reply:
Also, I see you only check the first account of the config. Does that mean that if the ledger account is not the first, it's completely ignored?